src/adapter/graphql/src/mutations/flows_mut/dataset_flow_configs_mut.rs

```diff
@@ -199,6 +198,14 @@ impl From<TimeDeltaInput> for chrono::Duration {
 
 ///////////////////////////////////////////////////////////////////////////////
 
+#[derive(InputObject)]
+struct BatchingConditionInput {
+    pub min_records_awaited: u64,
+    pub max_batching_interval: Option<TimeDeltaInput>,
```

Comment:
Should the interval be required? Perhaps even capped at 24 hours.

I think we should reserve batching for dealing with high-frequency data. If we allow infinite interval batching that would mean that certain transform flow can potentially live forever ... which may mess with event store compactions, or monitoring/alerting or hanged flows.

I was thinking that at least `max_interval` will be driving the watermark propagation.

Reply:
Agree. Made the field required, and added upper bound validation result.